pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -246,8 +247,8 @@ if [[ -e @out@/nix-support/cc-wrapper-hook ]]; then
 fi
 
 if (( "${NIX_CC_USE_RESPONSE_FILE:-@use_response_file_by_default@}" >= 1 )); then
-    responseFile=$(mktemp "${TMPDIR:-/tmp}/cc-params.XXXXXX")
-    trap 'rm -f -- "$responseFile"' EXIT
+    responseFile=$(PATH=$path_utils mktemp "${TMPDIR:-/tmp}/cc-params.XXXXXX")
+    trap 'PATH=$path_utils rm -f -- "$responseFile"' EXIT
```

Comment:
Why does PATH need to be explicitly set here again when it is already set in line 15/16?

Reply:
It is being restored to its original value in line 240.

Reply:
Yep, that's why this PR is needed. PATH restoration on line 240 is done for the hook a couple lines after. We could place PATH restoration after the hook and this response file-related code, but I'm not sure it won't break hooks somewhere.

Reply:
Ah right. Maybe we move that line inside the if and else instead of creating another variable?

Reply:
Although `cc-wrapper-hook` probably relies on the original PATH. So nvm.

Reply:
I think I found a better solution: instead of playing with `$PATH` variable we may just call `rm` and `mktemp` via absolute paths like `@coreutils_bin@/bin/mktemp`. Also it makes our intent to not use system paths much clearer. I'll try this and return :)

Reply:
Yep, that worked.